src/language-handlebars/utils.js

```diff
@@ -58,6 +58,18 @@ function isParentOfSomeType(path, types) {
   return isNodeOfSomeType(parentNode, types);
 }
 
+function findParentOfSomeType(path, types) {
+  let index = 0;
+  let parentNode = path.getParentNode(index);
+  while (parentNode && !isNodeOfSomeType(parentNode, types)) {
+    parentNode = path.getParentNode(++index);
+  }
+  if (parentNode) {
+    return index;
+  }
+  return -1;
+}
```

Comment:
I am a little concerned by the performance of the while loop because `parentNode && !isNodeOfSomeType(parentNode, types)` would be relatively lax. To address this, we could invert the condition to `parentNode && isNodeOfSomeType(parentNode, types)`. This is possible if we make the function more specific to our usecase. For instance, the function could be akin:
```js
function needsOppositeQuote(path) {
  let index = 0;
  let parentNode = path.getParentNode(index);
  while (parentNode && isNodeOfSomeType(parentNode, ["SubExpression"])) {
    ++index;
    parentNode = path.getParentNode(index);
  }

  if (
    isNodeOfSomeType(path.getParentNode(index + 1), ["ConcatStatement"])
    && isNodeOfSomeType(path.getParentNode(index + 2), ["AttrNode"])) {
    return true;
  }

  return false;
}
```

In that case, we could move the util function from the `utils.js` file (that holds generic utils) to the main `printer-glimmer.js` ([here for example](https://github.com/prettier/prettier/blob/7ced9e66549b8d5a8e3a74473af462f86fb0245c/src/language-handlebars/printer-glimmer.js#L692)).

wdyt?

Reply:
Good idea. I had the also concerns about the performance but was not aware of the fact all inside are SubExpressions.